src/plugins.js

```diff
@@ -0,0 +1,89 @@
+class BasePlugin {
+  static name = "untitled plugin";
+
+  registerDocumentFormats() {
+    return [];
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  parseDocument(contents, filename, format) {
+    return undefined;
+  }
+
+  registerOutputFormats() {
+    return [];
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  logSingleResult(result, filename, format) {
+    return undefined;
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  logAllResults(results, format) {
+    return undefined;
+  }
```

Comment:
Is "log" the right verb for these methods?

We're not actually logging or outputting anything here. We're returning a value which will then be logged by the calling code. Is there a better name for this? `getSingleResultMessage` and `getAllResultsMessage` might be more descriptive but less nice. Hmm :thinking:

Reply:
I decided to go with `getSingleResultLogMessage` and `getAllResultsLogMessage` here.
They are less snappy but more descriptive.

Naming things is hard.